tests/unit/test_derivative_utils.py

```diff
@@ -1,27 +1,110 @@
 
 
+def test_get_pipeline_from_backup_if_remote_fails(monkeypatch):
+    """
+    Test that the pipeline catalog is loaded from the local backup if the remote location is unreachable.
+    """
+    # TODO: Make a proper mock for the requests.get function
+    # or switch to httpx for better testing capabilities
+    nonsense_url = "https://does.not.exist.url"
+
+    def mock_httpx_get(*args, **kwargs):
+        response = httpx.Response(status_code=400, json={"key": "value"})
+        # This slightly odd construction is necessary to create a Response object
+        # that has the correct URL already baked in (I think), because otherwise we get the following
+        # RuntimeError: Cannot call `raise_for_status` as the request instance has not been set on this response.
+        # TODO: find a better solution or understand the problem better
+        response._request = httpx.Request("GET", nonsense_url)
+        return response
+
+    monkeypatch.setattr(httpx, "get", mock_httpx_get)
+
+    with pytest.warns(UserWarning) as w:
+        result = mappings.get_pipeline_catalog(
+            get_url=nonsense_url, get_path=mappings.PROCESSING_PIPELINE_PATH
+        )
+
+    assert all(isinstance(item, dict) for item in result)
+    assert "Unable to load pipeline catalog" in w[0].message.args[0]
+
+
+def test_raises_exception_if_remote_and_local_pipeline_catalog_fails(
+    monkeypatch, tmp_path
+):
+    """
+    If I cannot get the pipeline catalog from the remote location and the local backup, I should raise an exception.
+    """
+    # or switch to httpx for better testing capabilities
+    nonsense_url = "https://does.not.exist.url"
+
+    def mock_httpx_get(*args, **kwargs):
+        response = httpx.Response(status_code=400, json={"key": "value"})
+        # This slightly odd construction is necessary to create a Response object
+        # that has the correct URL already baked in (I think), because otherwise we get the following
+        # RuntimeError: Cannot call `raise_for_status` as the request instance has not been set on this response.
+        # TODO: find a better solution or understand the problem better
+        response._request = httpx.Request("GET", nonsense_url)
+        return response
+
+    monkeypatch.setattr(httpx, "get", mock_httpx_get)
+
+    with pytest.raises(FileNotFoundError) as e:
+        mappings.get_pipeline_catalog(
+            get_url=nonsense_url, get_path=tmp_path / "does_not_exist.json"
+        )
+
+    assert "Have you correctly initialized the submodules" in str(e.value)
+
+
+def test_get_pipeline_from_remote_succeeds(monkeypatch):
+    nonsense_url = "https://made.up.url/pipeline_catalog.json"
+    mock_pipeline_catalog = [
+        {"name": "sillypipe", "versions": ["1", "2"]},
+        {"name": "funpipe", "versions": ["10", "11"]},
+    ]
+
+    def mock_httpx_get(*args, **kwargs):
+        response = httpx.Response(status_code=200, json=mock_pipeline_catalog)
```

Comment:
same as above, I think we could use the `request=` param here instead of overriding it post-hoc.